### PR TITLE
Gurobi wrapper

### PR DESCRIPTION
**Pull request by anirudhSK**
Borrows several code snippets and ideas from https://github.com/ubnt-intrepid/rust-gurobi, but is considerably leaner.

### CODE REVIEW

**Comment by taegyunkim** [comment]
Taking a look.
Btw @anirudhSK test is failing, please take a look.

File: src/gurobi_ffi.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
+pub use std::os::raw::{c_int, c_double, c_char, c_void};
+pub type c_str = *const c_char;
```
Comment (taegyunkim) [inline src/gurobi_ffi.rs:2]: So this is just an alias for *const c_char type?

File: gurobi_example.c [hunk 2]
```diff
@@ -0,0 +1,144 @@
+/* Copyright 2019, Gurobi Optimization, LLC */
```
Comment (taegyunkim) [inline gurobi_example.c:1]: Does any other file use this? This is here for just reference right?

File: src/gurobi_ffi.rs (hunk 1 above)
Comment (anirudhSK) [reply src/gurobi_ffi.rs:2]: Yes.

File: gurobi_example.c (hunk 2 above)
Comment (anirudhSK) [reply gurobi_example.c:1]: Yes, it's just for reference. Let's keep it for a bit until we get comfortable with our Rust API to Gurobi.

**Comment by taegyunkim** [comment]
Looking at https://ci.appveyor.com/project/anirudhSK/dinarii/builds/27843727, the first failed appveyor run, it expects gurobi to be properly setup. 

I wonder whether there would be an easy way to setup gurobi license key with appveyor. 
This github repo also notes about proprietary licensing issue with gurobi and appveyor. 
https://github.com/google/or-tools/issues/903#issuecomment-433339895

**Comment by taegyunkim** [comment]
@anirudhSK After setting PATH and LD_LIBRARY_PATH, gurobi-sys builds. I think you need to set GRB_LICENSE_FILE environment variable to your license key file in local machine for gurobi_ffi to work, assuming that we continue to use running appveyor on your machine.

**Comment by anirudhSK** [comment]
I fixed the issue with the license file. Once you review it, we should be able to merge it.

**Review by taegyunkim — APPROVED** [review]

**Pull request merged**